File: python/twitter/followtolist/follow2list.rb
# -*- coding: utf-8 -*-

require "twitter"
require 'yaml'


def get_all_friends(client,screen_name)
  all_friends = []
  client.friend_ids(screen_name).each_slice(100).each do |slice|
    client.users(slice).each do |friend|
      all_friends << friend
    end
  end
  all_friends
end

def get_following(client,user)
   users = get_all_friends(client,user)
  #  users.each{|u| p u.screen_name}
   return users.map{|s|s.screen_name}
end

def addlist(client,users)
  list = client.create_list(users + "s_following" , options={mode:"public"})
  return list
end

twitter_token = YAML.load_file("auth.yaml")[1]
client = Twitter::REST::Client.new do |config|
  config.consumer_key        = twitter_token["consumer_key"]
  config.consumer_secret     = twitter_token["consumer_secret"]
end


user = gets
# followings = get_following(client,user)

p addlist(client,user)

File: python/twitter/followtolist/follow2list.py
#-*- encoding:utf-8
import urlparse
import oauth2 as oauth
import twitter,yaml
import re,dateutil.parser

def get_access_token(consumer_key,consumer_secret):
    request_token_url = 'https://api.twitter.com/oauth/request_token'
    access_token_url = 'https://api.twitter.com/oauth/access_token'
    authorize_url = 'https://api.twitter.com/oauth/authorize'
    consumer = oauth.Consumer(consumer_key, consumer_secret)
    client = oauth.Client(consumer)
    resp, content = client.request(request_token_url, "GET")
    if resp['status'] != '200':
        raise Exception("Invalid response %s." % resp['status'])
    request_token = dict(urlparse.parse_qsl(content))
    return request_token

auth = yaml.load(file("auth.yaml"))[0]
# oauth =  get_access_token(consumer_key,consumer_secret)

api = twitter.Api(auth["consumer_key"],
                  auth["consumer_secret"],
                  auth["access_token_key"],
                  auth["access_token_secret"]
                 )

from types import FunctionType

user = 'hide_clone'
def get_following(user):
    statuses =  api.GetFollowers(screen_name=user)

    # following = []
    # for t in statuses:
    #     print(t.screen_name)
    return map(lambda t:t.screen_name , statuses)
        

if __name__ == '__main__':
    user_name = raw_input('Enter positive user: ')
    # following = get_following(user_name)
    list = api.CreateList(user_name + "s_following",mode="private")
